src/mono/mono/mini/interp/transform.c

```diff
@@ -3770,28 +3787,43 @@ get_basic_blocks (TransformData *td, MonoMethodHeader *header, gboolean make_lis
 			break;
 		case MonoShortInlineBrTarget:
 			target = start + cli_addr + 2 + (signed char)ip [1];
+			if (target >= end) {
+				mono_error_set_generic_error (error, "System", "InvalidProgramException", "Jump target out of range at offset %04tx", cli_addr);
```

Comment:
Why does the method execution continues here if we detected error ?

Reply:
`get_basic_blocks` function is updated to return gboolean. It terminates the execution if invalid code is detected.